model/src/test/kotlin/licenses/LicenseConfigurationTest.kt

```diff
@@ -0,0 +1,144 @@
+            }
+        }
+
+        "detect duplicate license IDs" {
+            val lic1 = License(
+                SpdxSingleLicenseExpression.parse("ASL-1"), emptySortedSet(),
+                includeInNoticeFile = true, includeSourceCodeOfferInNoticeFile = false
+            )
+            val lic2 = License(
+                SpdxSingleLicenseExpression.parse("ASL-2"), emptySortedSet(),
+                includeInNoticeFile = true, includeSourceCodeOfferInNoticeFile = true
+            )
+            val lic3 = License(
+                SpdxSingleLicenseExpression.parse("ASL-1"), sortedSetOf("permissive"),
+                includeInNoticeFile = false, includeSourceCodeOfferInNoticeFile = true
+            )
+
+            shouldThrow<IllegalArgumentException> {
+                LicenseConfiguration(licenses = listOf(lic1, lic2, lic3))
+            }
+        }
+
+        "detect licenses referencing non-existing categories" {
+            val cat1 = LicenseCategory("Category 1")
+            val cat2 = LicenseCategory("Category 2")
+            val lic1 = License(
+                SpdxSingleLicenseExpression.parse("ASL-1"), sortedSetOf(cat1.name),
+                includeInNoticeFile = false, includeSourceCodeOfferInNoticeFile = true
+            )
+            val lic2 = License(
+                SpdxSingleLicenseExpression.parse("ASL-2"), sortedSetOf("unknownCategory"),
+                includeInNoticeFile = true, includeSourceCodeOfferInNoticeFile = false
+            )
+
+            shouldThrow<java.lang.IllegalArgumentException> {
+                LicenseConfiguration(licenseCategories = listOf(cat1, cat2), licenses = listOf(lic1, lic2))
+            }
+        }
+    }
+
+    "LicenseConfiguration" should {
+        "fetch all licenses for a specific category" {
+            val cat1 = LicenseCategory("permissive", "Permissive licenses")
+            val cat2 = LicenseCategory("non permissive", "Strict licenses")
+            val lic1 = License(
+                SpdxSingleLicenseExpression.parse("ASL-1"), sortedSetOf("permissive"),
+                includeInNoticeFile = true, includeSourceCodeOfferInNoticeFile = false
+            )
+            val lic2 = License(
+                SpdxSingleLicenseExpression.parse("ASL-2"), sortedSetOf("permissive"),
+                includeInNoticeFile = true, includeSourceCodeOfferInNoticeFile = true
+            )
+            val lic3 = License(
+                SpdxSingleLicenseExpression.parse("GPL"), sortedSetOf("non permissive"),
+                includeInNoticeFile = false, includeSourceCodeOfferInNoticeFile = true
+            )
+            val licenseConfiguration = LicenseConfiguration(
+                licenseCategories = listOf(cat1, cat2),
+                licenses = listOf(lic1, lic2, lic3)
+            )
+
+            val permissiveLicenses = licenseConfiguration.getLicensesForCategory(cat1.name)
+
+            permissiveLicenses shouldContainExactlyInAnyOrder setOf(lic1, lic2)
```

Comment:
consider instead `should containExactlyInAnyOrder(lic1, lic2)`
same in other palces.